uaaextras/webapp.py

```diff
@@ -322,15 +323,19 @@ def have_uaa_and_csrf_token():
                 None,
                 verify_tls=app.config["UAA_VERIFY_TLS"],
             )
+        
+        has_valid_token = False
 
         # if all looks good, setup the client
-        if token and token_checking_client.check_token_valid(token, app.config["CLIENT_ID"], app.config["CLIENT_SECRET"]):
+        if token:
             g.uaac = UAAClient(
                 app.config["UAA_BASE_URL"],
-                session["UAA_TOKEN"],
+                token,
                 verify_tls=app.config["UAA_VERIFY_TLS"],
             )
-        else:
+            has_valid_token = g.uaac.check_token_valid(token)
+
+        if not has_valid_token:
```

Comment:
because of the way we're mocking uaa, it really confused the tests to instantiate two clients in one function, so we're using the user's client to test its own token.